src/components/LoginComponent.vue

```diff
@@ -83,10 +80,18 @@ export default {
                         sessionStorage.setItem("authorizedUsername", this.username);
                         sessionStorage.setItem("token", result.access_token)
                         sessionStorage.setItem("userId", result.userId)
-                        sessionStorage.setItem("isLoggedIn", true)
-                        this.$store.dispatch('saveAuth');
                         this.$store.dispatch("saveSocket");
-                        this.$router.push("/posts")
+
+                        // Check if 2FA is enabled
+                        if (result.is2fa == true) {
```

Comment:
I think this we should not trust client. Client already has user details with this results.